### PR TITLE
add pool service controller.

### PR DESCRIPTION
**Pull request by zyjhtangtang**
#### What type of PR is this?

/kind feature
/sig network

#### What this PR does / why we need it:
The first-level controller for the multi-region load balancer.

#### Does this PR introduce a user-facing change?

```release-note
YES
```

### CODE REVIEW

File: pkg/yurtmanager/controller/poolservice/predicate.go [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+func isMultiRegionalLbService(svc *v1.Service) bool {
+	if svc.Spec.Type != v1.ServiceTypeLoadBalancer {
+		return false
+	}
+
+	if svc.Spec.LoadBalancerClass == nil {
+		return false
+	}
+
+	if svc.Annotations == nil {
```
[the remark this answers is not in the document]
Comment (zyjhtangtang) [reply pkg/yurtmanager/controller/poolservice/predicate.go:87]: If it is a multi-region lb, it must contain this annotations

File: pkg/yurtmanager/controller/poolservice/predicate.go [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+
+func isNodePoolSelectorChange(oldAnnotations, newAnnotations map[string]string) bool {
+	return !annotationValueIsEqual(oldAnnotations, newAnnotations, network.AnnotationNodePoolSelector)
+}
+
+func isServiceTypeChange(oldSvc, newSvc *v1.Service) bool {
+	return oldSvc.Spec.Type != newSvc.Spec.Type
+}
+
+func isFinalizersChange(oldSvc, newSvc *v1.Service) bool {
```
[the remark this answers is not in the document]
Comment (zyjhtangtang) [reply pkg/yurtmanager/controller/poolservice/predicate.go:133]: The finalizer was modified manually and the controller needs to add it again.

**Comment by rambohe-ch** [comment]
@zyjhtangtang I think the repo of controllers about MultiPoolLoadBalancer should be constructed as following:

```
controller:
|
-- loadbalancerset:
   |
   -- loadbalancerset:
   |
   -- enslb:
   | 
   -- viplb:
```

**Comment by zyjhtangtang** [comment]
> @zyjhtangtang I think the repo of controllers about MultiPoolLoadBalancer should be constructed as following:
> 
> ```
> controller:
> |
> -- loadbalancerset:
>    |
>    -- loadbalancerset:
>    |
>    -- enslb:
>    | 
>    -- viplb:
> ```

Thanks for the suggestion, I have adjusted the relevant directory structure

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go [hunk 3]
```diff
@@ -0,0 +1,513 @@
...
+		return errors.Errorf("resource %s isn't exist", nodepoolsControllerResource.String())
+	}
+
+	return add(mgr, r)
+}
+
+var _ reconcile.Reconciler = &ReconcilePoolService{}
+
+// ReconcilePoolService reconciles a PoolService object
+type ReconcilePoolService struct {
```
Comment (rambohe-ch) [inline pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go:91]: ReconcilePoolService --> ReconcileLoadBalancerSet

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go [hunk 4]
```diff
@@ -0,0 +1,513 @@
...
+	if err := r.addFinalizer(svc); err != nil {
+		return errors.Wrapf(err, "failed to add finalizer")
+	}
+
+	currentPoolServices, err := r.currentPoolServices(svc)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get current pool services for service %s/%s", svc.Namespace, svc.Name)
+	}
+
+	if err := r.aggregateToService(svc, currentPoolServices); err != nil {
```
Comment (rambohe-ch) [inline pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go:277]: aggregateToService should be removed out of reconcileNormal, and add a new function named reconcileService for updating service status and annotations.

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/finalizer.go [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	poolServiceFinalizer = "poolservices.openyurt.io/resources"
```
Comment (rambohe-ch) [inline pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/finalizer.go:27]: poolservices.openyurt.io/resources --> poolservice.openyurt.io/resources

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go [hunk 6]
```diff
@@ -0,0 +1,513 @@
...
+
+func Format(format string, args ...interface{}) string {
+	s := fmt.Sprintf(format, args...)
+	return fmt.Sprintf("%s: %s", names.LoadBalancerSetController, s)
+}
+
+// Add creates a new PoolService Controller and adds it to the Manager with default RBAC. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(ctx context.Context, c *appconfig.CompletedConfig, mgr manager.Manager) error {
+	klog.Infof(Format("poolservice-controller add controller %s", poolServicesControllerResource.String()))
```
Comment (rambohe-ch) [inline pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go:75]: poolservice-controller --> loadbalancerset-controller

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go [hunk 7]
```diff
@@ -0,0 +1,513 @@
...
+
+const (
+	labelManageBy = "poolservice.openyurt.io/managed-by"
+
+	poolServiceModifiedEventMsgFormat        = "PoolService %s/%s resource is manually modified, the controller will overwrite this modification"
+	poolServiceManagedConflictEventMsgFormat = "PoolService %s/%s is not managed by pool-service-controller, but the nodepool-labelselector of service %s/%s include it"
+)
+
+func init() {
+	flag.IntVar(&concurrentReconciles, "poolservice-workers", concurrentReconciles, "Max concurrent workers for PoolService controller.")
```
Comment (rambohe-ch) [inline pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go:58]: please refer to pull request https://github.com/openyurtio/openyurt/pull/1906 and improve concurrentReconciles definition.

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go [hunk 8]
```diff
@@ -0,0 +1,513 @@
...
+
+func (r *ReconcilePoolService) reconcileClean(svc *corev1.Service) error {
+	currentPoolServices, err := r.currentPoolServices(svc)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get current pool services for service %s/%s", svc.Namespace, svc.Name)
+	}
+
+	if canRemoveFinalizer(currentPoolServices) {
+		if err := r.removeFinalizer(svc); err != nil {
+			return errors.Wrapf(err, "failed to add finalizer")
```
Comment (rambohe-ch) [inline pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go:197]: failed to add finalizer --> failed to remove finalizer

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go [hunk 9]
```diff
@@ -0,0 +1,513 @@
...
+	}
+
+	if err := r.reconcileNormal(copySvc); err != nil {
+		return reconcile.Result{}, errors.Wrapf(err, "failed to reconcile service")
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func needCleanPoolServices(svc *corev1.Service) bool {
```
Comment (rambohe-ch) [inline pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go:177]: needCleanPoolServices --> shouldDeleteAllPoolServices

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go [hunk 10]
```diff
@@ -0,0 +1,513 @@
...
+	}
+
+	if !isLoadBalancerSetService(svc) {
+		return true
+	}
+
+	return false
+}
+
+func (r *ReconcilePoolService) reconcileClean(svc *corev1.Service) error {
```
Comment (rambohe-ch) [inline pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go:189]: reconcileClean --> DeleteAllPoolServices

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go [hunk 11]
```diff
@@ -0,0 +1,513 @@
...
+		}
+
+		if err := r.Delete(context.Background(), &ps); err != nil {
+			return errors.Wrapf(err, "failed to delete poolservice %s/%s", ps.Namespace, ps.Name)
+		}
+	}
+	return nil
+}
+
+func (r *ReconcilePoolService) reconcileNormal(svc *corev1.Service) error {
```
Comment (rambohe-ch) [inline pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go:267]: reconcileNormal --> reconcilePoolServices

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go [hunk 12]
```diff
@@ -0,0 +1,513 @@
...
+	for _, cps := range currentPoolServices {
+		if exist := r.isExistPoolService(desirePoolServices, cps); !exist {
+			deletions = append(deletions, cps)
+		}
+	}
+
+	return
+}
+
+func (r *ReconcilePoolService) isExistPoolService(poolServices []netv1alpha1.PoolService, ps netv1alpha1.PoolService) bool {
```
Comment (rambohe-ch) [inline pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go:423]: isExistPoolService --> isPoolServicePresent or DoesPoolServiceExist

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go [hunk 13]
```diff
@@ -0,0 +1,513 @@
...
+	}
+	return nil
+}
+
+func (r *ReconcilePoolService) reconcileNormal(svc *corev1.Service) error {
+	if err := r.addFinalizer(svc); err != nil {
+		return errors.Wrapf(err, "failed to add finalizer")
+	}
+
+	currentPoolServices, err := r.currentPoolServices(svc)
```
Comment (rambohe-ch) [inline pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go:272]: it is more readable to move `currentPoolServices, err := r.currentPoolServices(svc)` into `syncPoolServices `

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go [hunk 14]
```diff
@@ -0,0 +1,513 @@
...
+	}
+
+	if err := r.applyPoolServices(poolServicesToApply); err != nil {
+		return errors.Wrapf(err, "failed to apply pool services %v", poolServicesToApply)
+	}
+
+	return nil
+}
+
+func (r *ReconcilePoolService) desirePoolServices(svc *corev1.Service) ([]netv1alpha1.PoolService, error) {
```
Comment (rambohe-ch) [inline pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go:326]: desirePoolServices --> desiredPoolServices

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go [hunk 15]
```diff
@@ -0,0 +1,513 @@
...
+
+	npList := &v1beta1.NodePoolList{}
+	if err := r.List(context.Background(), npList, &client.ListOptions{LabelSelector: labelSelector}); err != nil {
+		return nil, err
+	}
+
+	return filterDeletionNodePools(npList.Items), nil
+}
+
+func filterDeletionNodePools(allItems []v1beta1.NodePool) []v1beta1.NodePool {
```
Comment (rambohe-ch) [inline pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go:359]: filterDeletionNodePools --> filterActiveNodePools

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go [hunk 16]
```diff
@@ -0,0 +1,513 @@
...
+	}
+
+	var pss []netv1alpha1.PoolService
+	for _, np := range nps {
+		pss = append(pss, buildPoolService(svc, &np))
+	}
+	return pss, nil
+}
+
+func (r *ReconcilePoolService) listNodePoolsWithService(svc *corev1.Service) ([]v1beta1.NodePool, error) {
```
Comment (rambohe-ch) [inline pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go:344]: listNodePoolsWithService --> listNodePoolsByLabelSelector

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go [hunk 17]
```diff
@@ -0,0 +1,513 @@
...
+	}
+
+	poolServiceList := &netv1alpha1.PoolServiceList{}
+	if err := r.List(context.TODO(), poolServiceList, listSelector); err != nil {
+		return nil, errors.Wrapf(err, "failed to list pool service with %s/%s", svc.Namespace, svc.Name)
+	}
+
+	currentPoolServices := filterInvalidPoolService(poolServiceList.Items)
+
+	return currentPoolServices, nil
```
Comment (rambohe-ch) [inline pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go:224]: return filterValidPoolService(poolServiceList.Items), nil

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go [hunk 18]
```diff
@@ -0,0 +1,513 @@
...
+}
+
+func filterInvalidPoolService(poolServices []netv1alpha1.PoolService) []netv1alpha1.PoolService {
```
Comment (rambohe-ch) [inline pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go:227]: filterInvalidPoolService --> filterValidPoolService

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller_test.go [hunk 19]
```diff
@@ -0,0 +1,968 @@
...
+)
+
+var (
+	elbClass = "elb"
+)
+
+func TestReconcilePoolService_Reconcile(t *testing.T) {
+	scheme := initScheme(t)
+
+	t.Run("test get service not found", func(t *testing.T) {
```
Comment (rambohe-ch) [inline pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller_test.go:61]: please add test case with `nodepool-labelselector` is empty.

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/event_handler.go [hunk 20]
```diff
@@ -0,0 +1,162 @@
...
+		DeleteFunc: func(deleteEvent event.DeleteEvent, limitingInterface workqueue.RateLimitingInterface) {
+			nodePoolRelatedServiceEnqueue(c, deleteEvent.Object, limitingInterface)
+		},
+		GenericFunc: func(genericEvent event.GenericEvent, limitingInterface workqueue.RateLimitingInterface) {
+			nodePoolRelatedServiceEnqueue(c, genericEvent.Object, limitingInterface)
+		},
+	}
+}
+
+func allMultiRegionLbServicesEnqueue(c client.Client, q workqueue.RateLimitingInterface) {
```
Comment (rambohe-ch) [inline pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/event_handler.go:118]: allMultiRegionLbServicesEnqueue --> allLoadBalancerSetServicesEnqueue

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/event_handler.go [hunk 21]
```diff
@@ -0,0 +1,162 @@
...
+	return
+}
+
+func NewNodePoolEventHandler(c client.Client) handler.EventHandler {
+	return handler.Funcs{
+		CreateFunc: func(createEvent event.CreateEvent, limitingInterface workqueue.RateLimitingInterface) {
+			allMultiRegionLbServicesEnqueue(c, limitingInterface)
+		},
+		UpdateFunc: func(updateEvent event.UpdateEvent, limitingInterface workqueue.RateLimitingInterface) {
+			handleNodePoolUpdate(c, updateEvent, limitingInterface)
```
Comment (rambohe-ch) [inline pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/event_handler.go:107]: handleNodePoolUpdate --> allLoadBalancerSetServicesEnqueue, don't need to verify labels change.

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/event_handler.go [hunk 22]
```diff
@@ -0,0 +1,162 @@
...
+	listSelector := client.MatchingLabels{
+		network.LabelNodePoolName: np.Name,
+		labelManageBy:             names.LoadBalancerSetController}
+
+	if err := c.List(context.Background(), poolServiceList, listSelector); err != nil {
+		return
+	}
+
+	for _, item := range poolServiceList.Items {
+		if !hasServiceName(&item) {
```
Comment (rambohe-ch) [inline pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/event_handler.go:157]: remove this if condition, because handlePoolServiceNormal has handled this condition.

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go (hunk 4 above)
Comment (zyjhtangtang) [reply pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go]: done

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/finalizer.go (hunk 5 above)
Comment (zyjhtangtang) [reply pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/finalizer.go]: done

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go (hunk 6 above)
Comment (zyjhtangtang) [reply pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go]: done

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go (hunk 7 above)
Comment (zyjhtangtang) [reply pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go]: done

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go (hunk 8 above)
Comment (zyjhtangtang) [reply pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go]: done

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go (hunk 9 above)
Comment (zyjhtangtang) [reply pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go]: done

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go (hunk 10 above)
Comment (zyjhtangtang) [reply pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go]: done

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go (hunk 11 above)
Comment (zyjhtangtang) [reply pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go]: done

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go (hunk 12 above)
Comment (zyjhtangtang) [reply pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go]: done

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go (hunk 13 above)
Comment (zyjhtangtang) [reply pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go]: done

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go (hunk 14 above)
Comment (zyjhtangtang) [reply pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go]: done

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go (hunk 17 above)
Comment (zyjhtangtang) [reply pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go]: done

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller_test.go (hunk 19 above)
Comment (zyjhtangtang) [reply pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller_test.go]: done

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/event_handler.go (hunk 20 above)
Comment (zyjhtangtang) [reply pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/event_handler.go]: done

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/event_handler.go (hunk 22 above)
Comment (zyjhtangtang) [reply pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/event_handler.go]: done

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go (hunk 15 above)
Comment (zyjhtangtang) [reply pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go]: filterDeletionNodePools means filtering out deleted nodepools

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go (hunk 16 above)
Comment (zyjhtangtang) [reply pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go]: done

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go (hunk 18 above)
Comment (zyjhtangtang) [reply pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go]: filterInvalidPoolService means filtering out invalid poolservices

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/loadbalancerset_controller.go [hunk 23]
```diff
@@ -0,0 +1,510 @@
...
+	copySvc := service.DeepCopy()
+
+	if shouldDeleteAllPoolServices(copySvc) {
+		if err := r.deleteAllPoolServices(copySvc); err != nil {
+			return reconcile.Result{}, errors.Wrapf(err, "failed to clean pool services")
+		}
+		return reconcile.Result{}, nil
+	}
+
+	if err := r.reconcileService(copySvc); err != nil {
```
Comment (rambohe-ch) [inline pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/loadbalancerset_controller.go:164]: reconcileService --> syncService, and move it behind reconcilePoolServices

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/loadbalancerset_controller.go [hunk 24]
```diff
@@ -0,0 +1,510 @@
...
+	isUpdatedLbStatus := compareAndUpdateServiceLbStatus(svc, lbStatus)
+
+	if !isUpdatedLbStatus && !isUpdatedAnnotations {
+		return nil
+	}
+
+	if err := r.Update(context.Background(), svc); err != nil {
+		return errors.Wrapf(err, "failed to update service")
+	}
+	return nil
```
Comment (rambohe-ch) [inline pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/loadbalancerset_controller.go:297]: return r.Update(context.Background(), svc)

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/loadbalancerset_controller.go [hunk 25]
```diff
@@ -0,0 +1,510 @@
...
+
+	if shouldDeleteAllPoolServices(copySvc) {
+		if err := r.deleteAllPoolServices(copySvc); err != nil {
+			return reconcile.Result{}, errors.Wrapf(err, "failed to clean pool services")
+		}
+		return reconcile.Result{}, nil
+	}
+
+	if err := r.reconcileService(copySvc); err != nil {
+		return reconcile.Result{}, errors.Wrapf(err, "failed to apply pool services")
```
Comment (rambohe-ch) [inline pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/loadbalancerset_controller.go:165]: failed to apply pool services --> failed to sync service(%s/%s), copySvc.Namespace, copySvc.Name

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/loadbalancerset_controller.go (hunk 23 above)
Comment (zyjhtangtang) [reply pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/loadbalancerset_controller.go:164]: done

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/loadbalancerset_controller.go (hunk 24 above)
Comment (zyjhtangtang) [reply pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/loadbalancerset_controller.go:297]: done

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/loadbalancerset_controller.go (hunk 25 above)
Comment (zyjhtangtang) [reply pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/loadbalancerset_controller.go:165]: done

**Review by rambohe-ch — APPROVED** [review]
